oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/query/input/Duration.java

```diff
@@ -27,45 +27,50 @@
 
 @Getter
 @Setter
+/**
+ * To optimize the query, the range of start and end times will be trimmed to [TTL_deadLine < time <= CurrentTime].
+ */
 public class Duration {
     private String start;
     private String end;
     private Step step;
 
     /**
-     * See {@link DurationUtils#convertToTimeBucket(Step, String)}
+     * See {@link DurationUtils#trimToStartTimeBucket(Step, String, boolean)}
      */
-    public long getStartTimeBucket() {
-        return DurationUtils.INSTANCE.convertToTimeBucket(step, start);
+    public long getStartTimeBucket(boolean isRecord) {
```

Comment:
When I review and read this method, I am wondering, whether we separate this method as two, is that better for reading?

Reply:
> When I review and read this method, I am wondering, whether we separate this method as two, is that better for reading?

Aggree, when adding a `boolean` argument to a method, it's usually better in terms of readability to split the method into two, like `getRecordStartTimeBucket` and `getXXXStartTimeBucket`